scripts/sanity_chk/harness.py

```diff
@@ -121,6 +130,50 @@ def handle(self, line):
 
         self.process_test(line)
 
+class Pytest(Harness):
+    def handle(self, line):
+        cmd = [
+			'pytest',
+			'-s',
+			os.path.join(self.source_dir, self.pytest_root),
+			'--cmdopt',
```

Comment:
what is this `--cmdopt` arg? I can't find it in the pytest docs and it causes an error when I try using some mockup test.

Reply:
please refer [here](https://docs.pytest.org/en/6.0.1/example/simple.html#pass-different-values-to-a-test-function-depending-on-command-line-options) for detail. to handle --cmdopt, in my case add a file conftest.py:
import pytest
  
   def pytest_addoption(parser):
       parser.addoption(
          '--cmdopt'
      )
 
  @pytest.fixture()
  def cmdopt(request):
      return request.config.getoption('--cmdopt')

Reply:
I am sorry, but I still don't understand how the workflow of adding the pytests should work. Do you mean that if someone wants to add his test and use sanitycheck harness he/she should also add a special config file with the code you put above? Shouldn't it be documented somewhere?

Reply:
Thank you @PerMac , updated, add some comment, sample code please refer https://github.com/youhuazhu/zephyr/commit/ad8decb8dcf48a307299ac840049efa7202ff598

Reply:
Thank you for providing the example. Could you please add the commit with the example to this PR? I think it is good to have this example within the same PR for consistency. It would also benefit the testing, as I could check its behavior within the scope of PR (being checked out on the PR).
IMO it would be perfect if you'd also direct to this test example in the docs section and explain the need for the extra functions required for handling `cmdopt` which the example is defining/using in its code.

Reply:
done